app/helpers/application_helper/button/middleware_server_action.rb

```diff
@@ -1,7 +1,18 @@
 class ApplicationHelper::Button::MiddlewareServerAction < ApplicationHelper::Button::Basic
 
   def visible?
-    !@record.present? ||
-      (@record.try(:product) != 'Hawkular' && @record.try(:middleware_server).try(:product) != 'Hawkular')
+    @record.present? && !hawkular_product? && !immutable?
```

Comment:
@cfcosta I agree with all of your changes, just not sure it should be part of this PR. I'm just adding an immutable field to existing infrastructure. And we need to get this merged quickly as this is a blocker issue for this release that the window is closing rapidly. Changing more lines of code will subject the PR to more scrutiny which could cause it to miss the release window.  You or I can add these changes to a follow-on refactoring PR immediately after this is merged.